src/components/stepforms/TopStepForm.vue

```diff
@@ -0,0 +1,63 @@
+<template>
+  <div>
+    <step-form-title :title="title"></step-form-title>
+    <WidgetInputText
+      id="limitInput"
+      v-model.number="editedStep.limit"
+      type="number"
+      name="Get top..."
+      placeholder="Enter a number of rows"
+    ></WidgetInputText>
+    <WidgetAutocomplete
+      id="rankOnInput"
+      v-model="editedStep.rank_on"
+      name="Sort column..."
+      :options="columnNames"
+      @input="setSelectedColumns({ column: editedStep.rank_on })"
+      placeholder="Select a column"
+    ></WidgetAutocomplete>
+    <WidgetAutocomplete
+      id="sortOrderInput"
+      v-model="editedStep.sort"
+      name="Sort order:"
+      :options="sortOrder"
+      placeholder="Select an order"
+    ></WidgetAutocomplete>
+    <WidgetMultiselect
+      id="groupbyColumnsInput"
+      v-model="editedStep.groups"
+      name="(Optional) Group by..."
+      :options="columnNames"
+      @input="setSelectedColumns({ column: editedStep.groups[editedStep.groups.length-1] })"
+      placeholder="Select columns"
+    ></WidgetMultiselect>
+    <step-form-buttonbar :errors="errors" :cancel="cancelEdition" :submit="submit"></step-form-buttonbar>
+  </div>
+</template>
+
+<script lang="ts">
+import { Prop } from 'vue-property-decorator';
+import { TopStep } from '@/lib/steps';
+import WidgetAutocomplete from '@/components/stepforms/WidgetAutocomplete.vue';
+import WidgetInputText from '@/components/stepforms/WidgetInputText.vue';
+import WidgetMultiselect from './WidgetMultiselect.vue';
+import BaseStepForm from './StepForm.vue';
+import { StepFormComponent } from '@/components/formlib';
+
+@StepFormComponent({
+  vqbstep: 'top',
+  name: 'top-step-form',
+  components: {
+    WidgetAutocomplete,
+    WidgetInputText,
+    WidgetMultiselect,
+  },
+})
+export default class TopStepForm extends BaseStepForm<TopStep> {
+  @Prop({ type: Object, default: () => ({ name: 'top', rank_on: '', sort: 'asc' }) })
+  initialStepValue!: TopStep;
+
+  readonly title: string = 'Top N rows';
+  sortOrder: TopStep['sort'][] = ['asc', 'desc'];
```

Comment:
We probably don't need a data prop for `sortOrder`: we don't need to modify it. I think we should just hardcode these option values in the widget.